JesusFilm/core
# feat: add rtl to journeys

# Description
[Basecamp Todos](https://3.basecamp.com/3105655/buckets/29688713/todolists/5413044373)

# How should this PR be QA Tested?

Make an RTL journey (eg locale is Arabic) in journeys-admin (use vercel link)
OR
Request access to premade journeys (if you want to edit)
LTR https://admin-stage.nextstep.is/journeys/3148f59e-9d40-4a52-aa5c-46830349f08a
RTL https://admin-stage.nextstep.is/journeys/7f28e898-d168-46a1-8614-d14080f63212
OR
Use premade journey direct link
https://your-stage.nextstep.is/ltr-journey-demo
https://your-stage.nextstep.is/rtl-journey-demo-2

Open journeys with journeys vercel link + slug of your own journey or premade journeys: https://journeys-git-22-08-zl-fix-conductor-rtl-jesusfilm.vercel.app/journeys/[slug]

Storybook should also show the correct styling. Check the `Conductor` stories and `Journey-ui` project stories
https://612c2a83fdc2b2003a5c2eb7-fniirshjkx.chromatic.com/?path=/story/journeys-conductor--default

Requirements:

**Swipe Navigation:**
- [x]  it should hide nav arrow on the **right** if on first card of journey
- [x]  it should be able to click on nav arrow on the **left** by default
- [x]  it should disable nav arrow on the **right** by default
- [x]  it should disable nav arrow on the **left** when next card is locked
- [x]  it should hide nav arrow on the **left** if on last card of journey
- [x]  it should right align the copyright notice

**RTL Journey Components**

**Typography**
- [x] Typography is right-aligned in RTL journeys
- [x]  RTL Typography to use the font [El Messiri](https://fonts.google.com/specimen/El+Messiri?query=el+messir#standard-styles)
- [x]  Typography font-size, margins, line-height and other styles adjusted for El Messiri. (excludes label, button, assistive) [Design](https://www.figma.com/file/4zMjrDyVVEgsSyWpJh9MgK/NS---Design-System?node-id=5%3A1013)
- [x]  Urdu theme uses Arial + it's own styling (excludes label, button, assistive)

**Card**
- [x]  Card content overlay displays on the **left** side on RTL journeys on contained layout on desktop
- [x]  Card content overlay **top/bottom edges** don't need to be flipped on mobile portrait and desktop
- [x]  Card content overlay **left/right edge** mirrored on mobile landscape
- [x]  Card content remains **centred** but text is left aligned on RTL journeys on expanded layout on desktop
- [x]  Scroll bar displays on **left** when content overflows

 
**Button**
- [x] it displays labels right-aligned in RTL journeys
- [x] it displays start icons on the right in RTL journeys
- [x] it displays end icons on the left in RTL journeys
- [x] it should use the RTL font style in RTL journeys for labels

**Multichoice**
- [x] it displays RTL text when the journey is RTL
- [x] it displays the checkboxes on the right when the journey is RTL
- [x] it uses the RTL font style in RTL journeys for labels

**TextField, SignUp, and TextResponse**
- [x]  it displays labels right-aligned in RTL journeys
- [x] it displays text value right-aligned in RTL journeys
- [x] it displays the error right align in RTL journeys
- [x] it displays the hint right aligned in RTL journeys

**Icons**
- [x]  it should enable users to add the ArrowBackRounded icon or Arrow Left on Admin
- [x]  it should enable users to add the ChevronLeftRounded icon
- [x]  it should render newly added icons ArrowBackRounded and ChevronLeftRounded

**Admin**
- [x]  it shows RTL card display on Editor Canvas
- [x]  it shows RTL cards on Control Panel
- [x]  it shows RTL cards Next Card attribute

# Checklist
